### PR TITLE
HSD8-724 Allow overriding node status of migrated items

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Allow users to unpublish nodes from migration and not re-publish those when migration runs again
- Fixed logic that deletes the record from the migration map table so if a node is deleted, it will be re-imported again.

# Need Review By (Date)
- 10/25

# Urgency
- Medium

# Steps to Test
1. checkout this branch
1. sync to mathematics site `blt drupal:sync --site=mathematics --partial`
1. go to `/node/17431/edit` and unpublish the node
1. run `drush @mathematics.local sqlq 'update migrate_map_hs_events_importer set hash = ""'`
1. run `drush @mathematics.local mim hs_events_importer`
1. verify `/node/17431` is still unpublished.
1. delete `/node/17431`
1. run `drush @mathematics.local mim hs_events_importer`
1. check `/admin/content` to verify the event was re-created.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr418-a1htsnmrorzkzvakpujcfhpzfaivarck.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db08c0544c5fa5b750e0a30

File: docroot/modules/humsci/hs_migrate/hs_migrate.module [hunk 1]
```diff
@@ -71,7 +71,14 @@ function hs_migrate_entity_delete(EntityInterface $entity) {
   foreach ($migrations as $migration) {
     $destination = $migration->get('destination');
 
-    if (isset($destination['plugin']) && strpos($destination['plugin'], 'entity:') !== FALSE) {
+    if (!isset($destination['plugin'])) {
```
Comment (sherakama) [inline docroot/modules/humsci/hs_migrate/hs_migrate.module:75]: Why would the destination plugin not be set?

Comment (sherakama) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module:75]: I'm trolling for a code comment here.

**Comment by sherakama** [comment]
Migration is packed with features. This is cool. Running manual tests now.

**Comment by sherakama** [comment]
As a quick aside. The httpd service keeps dying when put to sleep on that image.

File: docroot/modules/humsci/hs_migrate/hs_migrate.module (hunk 1 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module:75]: It should always be set. but its just a safety valve

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr418-a1htsnmrorzkzvakpujcfhpzfaivarck.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5db08c0544c5fa5b750e0a30

**Comment by sherakama** [comment]
```
[notice] Processed 3 items (0 created, 3 updated, 0 failed, 0 ignored) - done with 'hs_events_importer'
```

**Comment by sherakama** [comment]
```
 [notice] Processed 1 item (1 created, 0 updated, 0 failed, 0 ignored) - done with 'hs_events_importer'
```

**Pull request merged**